sdk/core/azure-core/azure/core/pipeline/_base_async.py

```diff
@@ -47,9 +48,9 @@ class _SansIOAsyncHTTPPolicyRunner(
     :type policy: ~azure.core.pipeline.policies.SansIOHTTPPolicy
     """
 
-    def __init__(self, policy: SansIOHTTPPolicy[HTTPRequestType, AsyncHTTPResponseType]) -> None:
+    def __init__(self, policy: SansIOHTTPPolicy[HTTPRequestType, HTTPResponseType]) -> None:
         super(_SansIOAsyncHTTPPolicyRunner, self).__init__()
-        self._policy = policy
+        self._policy: SansIOHTTPPolicy[HTTPRequestType, HTTPResponseType] = policy
```

Comment:
This shouldn't be necessary, as mypy/pyright will infer it from the constructor signature. Did you get a feedback from those tools?